tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/BuilderUtil.scala

```diff
@@ -38,16 +38,16 @@ object BuilderUtil {
     )(variable: TBuilderInstruction,
       set: TBuilderInstruction,
       expr: TBuilderInstruction): TBuilderInstruction = for {
-    usedBefore <- allUsed
+    usedBefore <- getAllUsed
     setEx <- set
```

Comment:
So, iiuc, this acccess to `set` is also changing the state, right? Then wouldn't it be clearer to have this something like:

```suggestion
    setEx <- getNextSet
```

Reply:
All `TBuilderInstruction`s "change state". The reason the parameters are named the way they are is to keep the intuition that, e.g. `builder.exists(x, S, p)` creates a `OperEx(exists, y, T, q)`-like expression, as not to bleed implementation details into the interface.

Reply:
This makes sense. So the set/get prefix can be saved just for the utilities that inspect the environment or update it, not for the operators that create the expressions. Thanks for explaining.